Packages/com.latios.latiosframework/Mimic/Mecanim/Internal/MecanimInternalUtilities.cs

```diff
@@ -458,11 +459,202 @@ internal static bool Approximately(float a, in float b)
+                                             ref SkeletonClipSetBlob clipSet,
+                                             NativeArray<MecanimParameter> parameters,
+                                             float deltaTime,
+                                             NativeList<TimedMecanimClipInfo> clipWeights,
+                                             DynamicBuffer<TimedMecanimClipInfo> previousFrameClipInfo,
+                                             float totalWeight,
+                                             float weightCullingThreshold)
+        {
+            //Get the current clip deltas
+            var currentDeltaTransform = TransformQvvs.identity;
+            //initialize scale and stretch to zero so that we can apply each clipweight additively
+            currentDeltaTransform.scale = 0f;
+            currentDeltaTransform.stretch = float3.zero;
+            for (int i = 0; i < clipWeights.Length; i++)
+            {
+                var clipWeight = clipWeights[i];
+                ref var clip = ref clipSet.clips[clipWeight.mecanimClipIndex];
+                var blendWeight = clipWeight.weight / totalWeight;
+
+                //Cull clips with negligible weight
+                if (blendWeight < weightCullingThreshold)
+                    continue;
+
+                ref var state = ref controllerBlob.layers[clipWeight.layerIndex].states[clipWeight.stateIndex];
+                var time = state.isLooping ? clip.LoopToClipTime(clipWeight.motionTime) : math.min(clipWeight.motionTime, clip.duration);
+                var stateSpeed = state.speedMultiplierParameterIndex != -1 ?
+                                 parameters[state.speedMultiplierParameterIndex].floatParam * state.speed :
+                                 state.speed;
+                var speedModifiedDeltaTime = deltaTime * stateSpeed;
+                var clipDeltaTransform = TransformQvvs.identity;
+                var hasLooped = state.isLooping && (time - speedModifiedDeltaTime < 0f || time + speedModifiedDeltaTime > clip.duration);
+
+                //If the clip has looped, get a sample of the end of the clip to incorporate it into the delta
+                if (hasLooped)
+                {
+                    clipDeltaTransform = clip.SampleBone(0, time);
+
+                    //Change times to account for negative speed
+                    var previousClipSampleTime = math.select(time - speedModifiedDeltaTime, clip.duration + time - speedModifiedDeltaTime, time - speedModifiedDeltaTime < 0f);
+                    var previousClipSample = clip.SampleBone(0, previousClipSampleTime);
+
+                    //Get the end of the clip
+                    var sampleEnd = math.select(clip.duration, 0, stateSpeed < 0f);
+                    var endClipSample = clip.SampleBone(0, sampleEnd);
+
+                    clipDeltaTransform.position += endClipSample.position - previousClipSample.position;
```

Comment:
I will ignore this for now, just because it implies a lot of math changes that also imply a lot of testing, but let's flag this as a test case for beta.  Testing negative state speeds with transitions can prove this out.